### PR TITLE
Omnicache bugfixes

### PR DESCRIPTION
**Pull request by joschock**
Resolves a couple of issues discovered when testing omnicache 0.11 at scale:

1. #264 
2. #263

### CODE REVIEW

File: edk2toolext/omnicache.py [hunk 1]
```diff
@@ -95,6 +95,11 @@ def _InitOmnicache(self):
         ret = RunCmd("git", "init --bare", workingdir=self.path)
         if (ret != 0):
             return ret
+
+        ret = RunCmd("git", "config --local fetch.parallel 0", workingdir=self.path)
```
Comment (corthon) [inline edk2toolext/omnicache.py:99]: Where are the details of these two config changes documented? Would it be appropriate to have a comment here about why the cache repo must be configured this way?

File: edk2toolext/omnicache.py [hunk 2]
```diff
@@ -145,6 +155,10 @@ def _ConvertOmnicache(self):
                 logging.info("Removing remote {0} with duplicate URL {1}".format(name, url))
                 RunCmd("git", "remote remove {0}".format(name), workingdir=self.path)
                 RunCmd("git", "config --local --unset omnicache.{0}.displayname".format(name), workingdir=self.path)
+        # setup fetch.parallel to a reasonable default
```
Comment (corthon) [inline edk2toolext/omnicache.py:158]: Why is this default reasonable

File: edk2toolext/omnicache.py [hunk 3]
```diff
@@ -393,6 +410,8 @@ def get_cli_options():
                        help="Update the Omnicache.  All cache changes also cause a fetch", default=False)
     group.add_argument("--no-fetch", dest="no_fetch", action="store_true",
                        help="Prevent auto-fetch if implied by other arguments.", default=False)
+    group.add_argument("--fetch-jobs", dest="fetch_jobs", type=int,
```
Comment (corthon) [inline edk2toolext/omnicache.py:413]: Is this the complement to "parallel" above?

**Review by corthon — APPROVED** [review]

File: edk2toolext/omnicache.py (hunk 1 above)
Comment (joschock) [reply edk2toolext/omnicache.py:99]: Will add a comment; the default behavior of git fetch is single threaded. fetch.parallel = 0 uses "a reasonable default" (per the git docs) - I looked into the source and presently that means a number of threads equal to current cpu count. Original implementation I set up forked a thread for each remote; this was running our build agent VMs out of memory.

File: edk2toolext/omnicache.py (hunk 2 above)
Comment (joschock) [reply edk2toolext/omnicache.py:158]: Will add a more informative comment; that was just the git verbiage.

File: edk2toolext/omnicache.py (hunk 3 above)
Comment (joschock) [reply edk2toolext/omnicache.py:413]: Yes; it allows the user to override the "reasonable" default of a thread per CPU and force more or less parallelization if needed. I added it for maximum flexibility here; but most implementations will likely just use the default.

File: edk2toolext/omnicache.py [hunk 4]
```diff
@@ -239,13 +255,16 @@ def UpdateRemote(self, oldUrl, newUrl=None, newName=None):
...
+    def Fetch(self, jobs=0):
         """Fetches all remotes"""
         logging.info("Fetching all remotes.")
         self._RefreshUrlLookupCache()
         # Tricky: we pass no-tags here, since we set up custom fetch refs for tags on a per-remote basis. This prevents
         # git from fetching the first set of tags into the global namespace.
-        return RunCmd("git", "fetch --all -j {0} --no-tags".format(len(self.urlLookupCache)), workingdir=self.path)
+        if (jobs != 0):
+            return RunCmd("git", "fetch --all -j {0} --no-tags".format(len(self.urlLookupCache)), workingdir=self.path)
+        else:
```
Comment (joschock) [inline edk2toolext/omnicache.py:266]: Bug: this should be the fetch-jobs argument. Will fix.

Comment (joschock) [reply edk2toolext/omnicache.py:266]: Fixed.

**Pull request merged**